src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+					//ShowContinueError( "  Iteration limit exceeded in calculating cold water mass flow rate" );
+				} else if ( SolFlag == -2 ) {
+					//ShowWarningError( "Cold water control failed in four pipe beam unit called " + this->name );
+					//ShowContinueError( "  Bad cold water flow limits" );
+				}
+				this->calc();
+				NonAirSysOutput = this->qDotBeamCooling;
+				return;
+			} else { // can run flat out without overcooling, which we just did
+				NonAirSysOutput = this->qDotBeamCooling;
+				return;
+			
+			}
+
+
+		} else if ( qDotBeamReq > DataHVACGlobals::SmallLoad && this->heatingAvailable ){ // beam heating needed
+			// first calc with max hot water flow
+			this->mDotCW = 0.0;
+			if ( this->beamCoolingPresent )
+			{
+				SetComponentFlowRate(this->mDotCW, this->cWInNodeNum, this->cWOutNodeNum, this->cWLocation.loopNum, 
+								this->cWLocation.loopSideNum, this->cWLocation.branchNum, this->cWLocation.compNum );
+			}
+			this->cWTempOut = this->cWTempIn;
+			this->mDotHW = this->mDotDesignHW;
+			this->calc();
+			if ( this->qDotBeamHeating > ( qDotBeamReq + DataHVACGlobals::SmallLoad ) ) {
+				this->qDotBeamHeatingMax = this->qDotBeamHeating;
+				// can overheat, modulate hot water flow to meet load
+				ErrTolerance = 0.01;
+				SolveRegulaFalsi( ErrTolerance, 50, SolFlag, this->mDotHW, std::bind( &HVACFourPipeBeam::residualHeating, this , _1), 0.0, this->mDotDesignHW);
+				if ( SolFlag == -1 ) {
+					//ShowWarningError( "Hot water control failed in four pipe beam unit called " + this->name );
+					//ShowContinueError( "  Iteration limit exceeded in calculating hot water mass flow rate" );
+				} else if ( SolFlag == -2 ) {
+					//ShowWarningError( "Hot water control failed in four pipe beam called " + this->name );
+					//ShowContinueError( "  Bad hot water flow limits" );
+				}
+				this->calc();
+				NonAirSysOutput = this->qDotBeamHeating;
+				return;
+			
+			} else { // can run flat out without overheating, which we just did
+				NonAirSysOutput = this->qDotBeamHeating;
+				return;
+			
+			}
+
+		} else {
+			this->mDotHW = 0.0;
+			if ( this-> beamHeatingPresent )
+			{
+				SetComponentFlowRate(this->mDotHW, this->hWInNodeNum, this->hWOutNodeNum, this->hWLocation.loopNum, 
+								this->hWLocation.loopSideNum, this->hWLocation.branchNum, this->hWLocation.compNum );
+			}
+			this->hWTempOut = this->hWTempIn;
+			// assume if there is still flow that unit has an internal bypass and convector does not still heat
+			this->mDotCW = 0.0;
+			this->cWTempOut = this->cWTempIn;
+			if ( this -> beamCoolingPresent)
+			{
+				SetComponentFlowRate(this->mDotCW, this->cWInNodeNum, this->cWOutNodeNum, this->cWLocation.loopNum, 
+								this->cWLocation.loopSideNum, this->cWLocation.branchNum, this->cWLocation.compNum );
+			}
+			// assume if there is still flow that unit has an internal bypass and convector does not still cool
+			// don't even need to run calc
+			return;
+		}
+
+		return;
```

Comment:
@EnergyArchmage You don't really need this return here, but it doesn't hurt anything.